src/scenic/core/vectors.py

```diff
@@ -4,427 +4,659 @@
-				return instance
-			elif any(needsSampling(arg) for arg in args):
-				return VectorMethodDistribution(method, instance, args, {})
-			elif any(needsLazyEvaluation(arg) for arg in args):
-				# see analogous comment in distributionFunction
-				return makeDelayedFunctionCall(helper, args, {})
-			else:
-				return wrapped(instance, *args)
-		return helper(*args)
-	return wrapper(method)
+    """Decorator for vector operators that yield vectors."""
+    op = method.__name__
+    setattr(VectorDistribution, op, makeVectorOperatorHandler(op))
+
+    @functools.wraps(method)
+    def helper(self, *args):
+        if needsSampling(self):
+            return VectorOperatorDistribution(op, self, args)
+        elif preservesZero and all(coord == 0 for coord in self.coordinates):
+            return self
+        elif any(needsSampling(arg) for arg in args):
+            return VectorMethodDistribution(method, self, args, {})
+        elif any(needsLazyEvaluation(arg) for arg in args):
+            # see analogous comment in distributionFunction
+            return makeDelayedFunctionCall(helper, args, {})
+        else:
+            return method(self, *args)
+    return helper
+
 def zeroPreservingVectorOperator(method):
-	return vectorOperator(method, preservesZero=True)
+    return vectorOperator(method, preservesZero=True)
 
 def vectorDistributionMethod(method):
-	"""Decorator for methods that produce vectors. See distributionMethod."""
-	@decorator.decorator
-	def wrapper(wrapped, instance, *args, **kwargs):
-		def helper(*args, **kwargs):
-			if any(needsSampling(arg) for arg in itertools.chain(args, kwargs.values())):
-				return VectorMethodDistribution(method, instance, args, kwargs)
-			elif any(needsLazyEvaluation(arg)
-			         for arg in itertools.chain(args, kwargs.values())):
-				# see analogous comment in distributionFunction
-				return makeDelayedFunctionCall(helper, args, kwargs)
-			else:
-				return wrapped(instance, *args, **kwargs)
-		return helper(*args, **kwargs)
-	return wrapper(method)
+    """Decorator for methods that produce vectors. See distributionMethod."""
+    @functools.wraps(method)
+    def helper(self, *args, **kwargs):
+        if any(needsSampling(arg) for arg in itertools.chain(args, kwargs.values())):
+            return VectorMethodDistribution(method, self, args, kwargs)
+        elif any(needsLazyEvaluation(arg)
+                 for arg in itertools.chain(args, kwargs.values())):
+            # see analogous comment in distributionFunction
+            return makeDelayedFunctionCall(helper, (self,) + args, kwargs)
+        else:
+            return method(self, *args, **kwargs)
+    return helper
+
+class Orientation:
+    """An orientation in 3D space."""
+    def __init__(self, rotation):
+        if not isinstance(rotation, Rotation):
+            raise ValueError("Orientation's 'rotation' parameter must be a SciPy rotation."\
+                " Perhaps you want to use a constructor?")
```

Comment:
This should be a `TypeError`. Also I would not use the term "constructor" since Python doesn't have those; "factory method" would be clearer. (Although since none of these functions should be called directly from Scenic programs, I don't know if we need to make any suggestions at all. Probably we should expand the docstring for this class with an `Args:` section to cover the initializer, and add docstrings to `fromQuaternion` and `fromEuler` so that people working on Scenic internals can see the available ways to create an `Orientation`.)